qa/rpc-tests/wallet-dump.py

```diff
@@ -96,12 +114,21 @@ def run_test (self):
         self.nodes[0].walletpassphrase('test', 10)
         # Should be a no-op:
         self.nodes[0].keypoolrefill()
-        self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.encrypted.dump")
+        key = None
+        try:
+            self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.encrypted.dump")
+        except Exception as ex:
+            key = get_dumpwallet_otp (ex.error['message'])
+            self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.encrypted.dump", key)
+        assert key, 'Import wallet did not raise exception when was called first time without one-time code.'
 
         found_addr, found_addr_chg, found_addr_rsv, hd_master_addr_enc = \
             read_dump(tmpdir + "/node0/wallet.encrypted.dump", addrs, hd_master_addr_unenc)
         assert_equal(found_addr, test_addr_count)
-        assert_equal(found_addr_chg, 90 + 1 + 50)  # old reserve keys are marked as change now
+        
+        # - old reserve keys are marked as change now
+        # - As wallet encryption creates a new master seed, it adds another set of hdmint keys.
+        assert_equal(found_addr_chg, 90 + 1 + 50 + (hdmint_key_count * 2))
```

Comment:
@riordant same question here.

Reply:
@ultimaweapon what question is that? Your last comment seems unrelated

Reply:
HD mint keys are not added to the reserve pool, that is why they don't show as reserve keys. You are correct in that they probably shouldn't show `change` keys however. I could add a `hdmint=1` value for these?

Reply:
I see. I think we can do it later. Already created an issue: https://github.com/zcoinofficial/zcoin/issues/602